lib/HotKeysHOC.js

```diff
@@ -1 +1,79 @@
-// TODO: Expose HotKeys functionality as HoC
\ No newline at end of file
+import React, {PureComponent} from 'react';
+import HotKeys from './HotKeys';
+import _ from 'lodash';
+
+const DEFAULT_STATE = {
+  keyboardAction: null,
+  keyboardEvent: null,
+};
+
+/**
+* HotKeysHOC is an HOC that provides the wrappedComponent with the ability to implement keyboard actions
+* without the user wrapping every component with a <HotKeys> component individually
+*
+* See examples/master/HOCWrappedNode.js for an example implementation
+* Follow the steps below to use the HOC:
+*
+* 1. Declared an ACTION_KEY_MAP
+* // example action to key map for a dropdown menu
+* const HIDE_CHILD = 'hideChild';
+* const SHOW_CHILD = 'showChild';
+*
+* const ACTION_KEY_MAP = {
+*     [HIDE_CHILD]: 'esc',
+*     [SHOW_CHILD]: ['down', 'space', 'enter']
+* };
+*
+* 2) Add or amend `componentWillReceiveProps({keyboardAction, keyboardEvent})`
+*
+* Declare let variable `keyboardActionHelper` with keys of the ACTION_KEY_MAP
+* Object keys for keyboardActionHelper === Object keys for ACTION_KEY_MAP
+* Values for keyboardActionHelper are handler methods that the user invokes upon keyboardAction
+*
+* Add the code snippet below to the end of `componentWillReceiveProps`:
+* if (_.has(keyboardActionHelper, keyboardAction)) {
+*     keyboardActionHelper[keyboardAction]();
+* }
+*
+* 3) Wrap the Component with withKeyboardSupport
+* - Remove `export default` from Component class definition'
+* - Insert `export default withKeyboardSupport(ACTION_KEY_MAP)(<insert_Component_name>);` at the end of `Component.js`
+* - Alternatively, use `const WrappedApp = HotKeysHOC(ACTION_KEY_MAP)(<componentName>);` to wrap a component
+*
+* @summary An HOC that provides the wrappedComponent with the ability to implement keyboard actions
+*/
+const HotKeysHOC = (keyMap) => ((Component) =>
+    class HotKeysHOCWrapper extends PureComponent {
+      static propTypes = Component.propTypes;
+      state = DEFAULT_STATE;
+
+      // Reset the state so keyboardActions and events do not linger on the HOC state
+      componentDidUpdate() {
+        this.setState(DEFAULT_STATE);
+      }
+
+      _setAccessibilityState(keyboardAction, keyboardEvent) {
+        this.setState({keyboardAction, keyboardEvent});
+      }
+
+      _getBoundHandlers() {
+        return _.mapValues(keyMap, (keyCombination, keyboardAction) => this._setAccessibilityState.bind(this, keyboardAction));
+      }
+
+      render() {
+        let {keyboardAction, keyboardEvent} = this.state;
+
+        return (
+            <HotKeys component="documentFragment" keyMap={keyMap} handlers={this._getBoundHandlers()}>
```

Comment:
Perhaps I am missing something but I am not sure why you have chosen `documentFragment` as the component to use, here.

Reply:
This is to avoid applying unexpected stylistic changes to the wrapped component that could come from `div`s.